src/app/book/books-all/books-all.component.ts

```diff
@@ -35,6 +37,28 @@ export class BooksAllComponent implements OnInit {
     )
   }
 
+  public deleteBook(id: number): void {
+
+    this.idToBeDeleted = id;
+
+  }
+
+  public changeModalVisible() {
+    this.isModalVisible = true;
+  }
+
+  public confirm() {
+    console.log(this.idToBeDeleted);
+    this.bookService.deleteBook(this.idToBeDeleted).subscribe(
+      (response: any) => {
```

Comment:
Remove empty body function and catch the error with something like this. `.pipe(catchError(this.errorHandler))`